src/basic/fd-util.c

```diff
@@ -822,6 +822,25 @@ int fd_reopen(int fd, int flags) {
         return new_fd;
 }
 
+int fd_reopen_propagate_flags(int fd, int flags, int propagate_flags) {
+        /* Invokes fd_reopen(fd, flags),
+         * but propagates selected flags if set on the original fd.
+         *
+         * You should use this for example if the original fd potentially is O_APPEND,
+         * otherwise we get rather "unexpected" behavior.
+         * Unless you intentionally want to overwrite pre-existing data,
+         * and have your output overwritten by the next guy.
+         *
+         * Used for "systemd-run --pty >> some-log", but may be used
+         * to propagate other flags as well, if there is a use case. */
+
+        int orig_status_flags = fcntl(fd, F_GETFL);
```

Comment:
so you opted to propagate the O_APPEND flag (which is a really weird flag i think, given its atomic-but-not-really-atomic behaviour), which works for your case I guess, but i am not sure it's the best approach to this. I think it would be nicer to copy the file offset over instead. It will cover your specific case but generically solve at the same time any case where people did not use O_APPEND, but did use an offset. Thus, it's a more comprehensive solution.

Reply:
also, if the fhcntl() fail here, just return a failure (i.e. `return -errno`). It's one of the most basic syscalls, and if it fails something is seriously wrong, there's no point in carrying on.

Reply:
propagating the file offset rather than O_APPEND also has the benefit that it does something reasonable for stdin, too.

Reply:
O_APPEND is not atomic on NFS, and supposedly not on writes above 4K in size though, afaik and a google search confirms. That's what i meant with "atomic-but-not-atomic". It's a bit too vague to rely on. it might work if you know your backing fs, and the size of writes you will actually make, but this is not so clear here. It's a hope, but not a fact.

Reply:
Actually I think it's OK to use O_APPEND here. In most cases we get the desired behavior without introducing something very specific. People who really want this to work on NFS probably can also use `--pipe`?

Reply:
i mean, if yo want to keep O_APPEND, i'd be fine with that. but make the interface more specific then, as the inheritance only works in case of O_APPEND and not generically for all O_XYZ flags, as mentioned elsewhere.